traffic/traffic_info.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "traffic/speed_groups.hpp"
+
+#include "indexer/mwm_set.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/map.hpp"
+#include "std/vector.hpp"
+
+namespace traffic
+{
+// This class is responsible for providing the real-time
+// information about road traffic for one mwm file.
+class TrafficInfo
+{
+public:
+  struct RoadSegmentId
+  {
+    // The ordinal number of feature this segment belongs to.
+    uint32_t m_fid;
+
+    // The ordinal number of this segment in the list of
+    // its feature's segments.
+    uint16_t m_idx : 15;
+
+    // The direction of the segment.
+    // todo(@m) Write up what the values 0 and 1 mean specifically.
+    uint8_t m_direction : 1;
+
+    bool operator<(RoadSegmentId const & o) const
+    {
+      if (m_fid != o.m_fid)
+        return m_fid < o.m_fid;
+      if (m_idx != o.m_idx)
+        return m_idx < o.m_idx;
+      return m_direction < o.m_direction;
+    }
+  };
+
+  TrafficInfo() = default;
+
+  TrafficInfo(MwmSet::MwmId const & mwmId);
+
+  // todo(@m) unordered_map?
+  using Coloring = map<RoadSegmentId, SpeedGroup>;
+
+  // Fetches the latest traffic data from the server and updates the coloring.
+  // todo(@m, @syershov) Currently a hardcoded path is used.
+  // Construct the url by passing an MwmValue.
```

Comment:
MwmValue won't help you to construct the url. CountryFile or LocalCountryFile may help you.